src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -283,17 +253,12 @@ def drop_application_before_upgrade(
             else:
                 generic_sql_error_handler(err)
 
-    def upgrade_app(
+    def create_or_upgrade_app(
         self,
         policy: PolicyBase,
-        is_interactive: bool,
-        version: Optional[str] = None,
-        patch: Optional[int] = None,
+        install_method: SameAccountInstallMethod,
```

Comment:
I agonized for a bit over defaulting this to either unversioned_dev or release_directive, but decided to make the call sites explicit instead. It does add to the test noise somewhat but they're more explicit now.

Reply:
yeah that sounds like a fair choice, I'm not convinced there is a suitable default either.